src/api/getBaladyiatsForDaira.js

```diff
@@ -0,0 +1,40 @@
+const MIN_DAIRA_LENGTH = 3;
+
+const getBaladyiatsForDaira = data =>
+/**
+ * Takes a wilaya daira name and returns its baladyiats
+ *
+ * @example get baladyiats for daira
+ * getBaladyiatsForDaira("ORAN")
+ *
+ * @param { String } daira The Wilaya's "daira"
+ * @returns { object[] | null } Returns All baladyiats for daira, or null
+ */
+
+  (daira) => {
+    if (!daira || daira.trim().length < MIN_DAIRA_LENGTH) {
+      return null;
+    }
+
+    const dairaName = daira.toLowerCase();
+
+    const wilaya = data.find(w =>
+      w.dairats.find(d =>
+        d.name.toLowerCase() === dairaName
+      || d.name_en.toLowerCase() === dairaName
+      || d.name_ar === dairaName
+      )
+    ) || null;
```

Comment:
Not needed, as [`.find`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) returns either the element or `undefined`

```suggestion
    );
```